set `NiceMonomorphism` info on the GAP side

ThomasBreuer opened the pull request:
Once a `isomorphic_group_over_finite_field` has been computed, the mapping between the two Oscar groups can now be used for the underlying GAP groups.
For that, we store it as a `NiceMonomorphism` in the characteristic zero GAP group.

I have changed the test behaviour a little:
`isomorphic_group_over_finite_field` computes already the group order for the isomorphic group over the finite field.
Therefore it does not make sense to test that the group in characteristic zero does not know its `order` value before an explicit `order` call. Instead we set now the order already in `isomorphic_group_over_finite_field`.
(The situation was not really bad before, the `order` call triggered `isomorphic_group_over_finite_field` again, but the value was already cached.)

This addresses an item in the discussion of #1779.

ThomasBreuer committed:
set `NiceMonomorphism` info on the GAP side

Once a `isomorphic_group_over_finite_field` has been computed,
the mapping between the two Oscar groups can now be used for the
underlying GAP groups.
For that, we store it as a `NiceMonomorphism` in the
characteristic zero GAP group.

lgoettgens commented:
I am not quite sure why that's happening, but all of the CI jobs with the `short` testset fail. After stripping away everything from other threads and just threading-related printing, I am left just with:
```julia
Starting tests for /home/runner/work/Oscar.jl/Oscar.jl/test/InvariantTheory/secondary_invariants.jl
Package Oscar errored during testing (received signal: 11)
Error: Process completed with exit code 1.
```
which does not really help with debugging, but at least we have a testfile to run separately to track down the problem.

ThomasBreuer commented:
Thanks @lgoettgens .
I can reproduce an error when I run the tests from that file.

ThomasBreuer committed:
fix the Julia functions used by GAP, improve the test

That was subtle:
By default, the Julia function used for embedding a matrix
into a matrix group makes a membership test.
We use the embedding in order to write the functions that map
characteristic zero matrices in GAP to matrices over finite fields
in GAP and vice versa.
These functions describe the `NiceMonomorphism` of the
characteristic zero matrix group in GAP,
hence they implement in particular the membership test on the GAP side.
Thus we have to omit the membership tests on the Julia side
in order to avoid infinite recursions.
